gpytorch/lazy/added_diag_lazy_tensor.py

```diff
@@ -125,3 +129,21 @@ def _init_cache_for_non_constant_diag(self, eye, batch_shape, n):
         logdet = self._r_cache.diagonal(dim1=-1, dim2=-2).abs().log().sum(-1).mul(2)
         logdet -= (1.0 / self._noise).log().sum([-1, -2])
         self._precond_logdet_cache = logdet.view(*batch_shape) if len(batch_shape) else logdet.squeeze()
+
+    @cached(name="svd")
+    def _svd(self) -> Tuple["LazyTensor", Tensor, "LazyTensor"]:
+        diag = self._diag_tensor.diag()
+        if torch.equal(diag, diag[..., :1].expand(diag.shape)):
+            U, S_, V = self._lazy_tensor.svd()
+            S = S_ + diag  # this assumes all diagonal entries are positive
+            return U, S, V
+        return super()._svd()
+
+    @cached(name="symeig")
+    def _symeig(self, eigenvectors: bool = False) -> Tuple[Tensor, Optional[LazyTensor]]:
+        diag = self._diag_tensor.diag()
+        if torch.equal(diag, diag[..., :1].expand(diag.shape)):
```

Comment:
```suggestion
        if torch.equal(diag, diag[..., :1].expand(diag.shape)) and torch.all(diag[..., :1] >= 0):
```
Let's just explicitly check that all the diagonal elements are positive insteaad of assuming it.